File: 11. Container With Most Water.cpp
class Solution2 {
public:
	int maxArea(vector<int>& height) {
		int i = 0, j = height.size() - 1;
		int area = -1;
		while(i < j){
			area = max(area, min(height[i], height[j]) * (j - i));
			height[i] < height[j] ? ++i : --j;
		}
		return area;
	}
};

File: 11. Container With Most Water.py
class Solution(object):

    def maxArea(self, height):

        """

        :type height: List[int]

        :rtype: int

        """

        left, right = 0, len(height)-1

        max_size = (right-left) * min(height[left], height[right])

        while(left<right):

            if height[left]<height[right]:

                left += 1

            else:

                right -= 1

            max_size = max(max_size, (right-left) * min(height[left], height[right]))

        return max_size
